### PR TITLE
HPCC-15925 Minor code generator speed ups and memory reduction

### PR DESCRIPTION
**Pull request by ghalliday**

### CODE REVIEW

**Comment by ghalliday** [comment]
@shamser please review

File: ecl/hql/hqlattr.cpp [hunk 1]
```diff
@@ -2550,9 +2564,10 @@ IHqlExpression * calcRowInformation(IHqlExpression * expr)
...
             retrieveRowInformation(info, ds);
 
-            __int64 limit = getIntValue(expr->queryChild(1), 0);
+            IHqlExpression * limitExpr = expr->queryChild(1);
+            __int64 limit = getIntValue(limitExpr, 0);
             if ((limit != 0) && !isGrouped(expr))
-                info.applyChoosen(limit, isLocalActivity(expr));
+                info.applyChoosen(limitExpr, limit, isLocalActivity(expr));
             else
                 info.limitMin(limit);
```
Comment (shamser) [inline ecl/hql/hqlattr.cpp]: Would it be possible to common up this code? 

i.e. replace lines 2568-2572 with
info.applyChoosen(limitExpr, isLocalActivity(expr), !isGrouped(expr))

and 
move the "if ((limit != 0) && !isGrouped(expr))" to within applyChoosen()?

**Comment by richardkchapman** [comment]
@ghalliday This is back to you at present

**Comment by ghalliday** [comment]
@shamser rebased and change from code review added.

**Comment by shamser** [comment]
@ghalliday Looks good.

**Pull request merged**